Makefile

```diff
@@ -0,0 +1,49 @@
+# Run `make help` to display help
+
+# --- Global -------------------------------------------------------------------
+
+all: check format ## check and format
+	@if [ -e .git/rebase-merge ]; then git --no-pager log -1 --pretty='%h %s'; fi
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+check:  ## Run shellcheck and shfmt
+	shellcheck *.sh
+	shfmt -i 4 -d *.sh
+
+format:  ## Format shell scripts.
+	shfmt -i 4 -w *.sh
+
+# --- Release -------------------------------------------------------------------
+NEXTTAG := $(shell { git tag --list --merged HEAD --sort=-v:refname; echo v0.0.0; } | grep -E "^v?[0-9]+.[0-9]+.[0-9]+$$" | head -n1 | awk -F . '{ print $$1 "." $$2 "." $$3 + 1 }')
+
+release: ## Tag release
+	git tag $(NEXTTAG) -a
+	git push origin $(NEXTTAG)
+
+.PHONY: release
+
+# --- Utilities ----------------------------------------------------------------
+COLOUR_NORMAL = $(shell tput sgr0 2>/dev/null)
+COLOUR_RED    = $(shell tput setaf 1 2>/dev/null)
+COLOUR_GREEN  = $(shell tput setaf 2 2>/dev/null)
+COLOUR_WHITE  = $(shell tput setaf 7 2>/dev/null)
+
+help:
+	@awk -F ':.*## ' 'NF == 2 && $$1 ~ /^[A-Za-z0-9%_-]+$$/ { printf "$(COLOUR_WHITE)%-25s$(COLOUR_NORMAL)%s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
+
+$(O):
+	@mkdir -p $@
+
+define nl
```

Comment:
rather than defining nl (which can be useful for multiple reasons, you could instead define the error message as a multi-line var:
```
define unsupported_make
Unsupported Make version.
Use GNU Make 3.82 or higher (current: $(MAKE_VERSION)).
Activate 🐚 hermit with `. bin/activate-hermit` and run again
endef

if ($(filter undefine,$(value .FEATURES)),)
$(error $(unsupported_make))
endif
```

Reply:
removed.